### PR TITLE
New feature: --skip-path

### PR DESCRIPTION
**Pull request by andrewjhumphrey**
Whilst monkeying about with AWS SSO we discovered it creates roles and
doesn't tag them with anything that we can use to discriminate on with existing
iamy skip functionality.  It does however use a path of `/aws-reserved/`
for all of it's autogenerated roles.

This PR adds the new flag --skip-path and will exclude from consideration
any resource that has a path that starts with any of the specified unwanted 
paths.

This will let us ignore the AWS SSO created roles if we want to.

I've added a test for this that I think is sufficient, but I'm interested
to know if folks think we need to add all of the edge cases to make sure
that strings.HasPrefix is working correctly.

The other question I had was is adding to the isSkippableManagedResource
function the right approach?  Perhaps splitting into a isSkippablePath,
isSkippableTag and isSkippableResourceName function and making the top 
level function just a wrapper would be easier to test deeply?  I'm
not convinced it is worth it, but love some feedback...

### CODE REVIEW

File: iamy.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ func main() {
 		skipCfnTagged   = kingpin.Flag("skip-cfn-tagged", fmt.Sprintf("Shorthand for --skip-tagged %s", cloudformationStackNameTag)).Bool()
 		skipTagged      = kingpin.Flag("skip-tagged", "Skips IAM entities (or buckets associated with bucket policies) tagged with a given tag").Strings()
 		includeTagged   = kingpin.Flag("include-tagged", "Includes IAM entities (or buckets associated with bucket policies) tagged with a given tag").Strings()
+		skipPaths       = kingpin.Flag("skip-path", fmt.Sprintf("Skips IAM entities that have a path starting with the supplied prefixes, repeat flag for multiple prefixes")).Strings()
```
Comment (liamdawson) [inline iamy.go]: If we want to take this approach (specifically handling prefixes), I'd suggest we try to align with a convention of some kind. I notice `aws iam list-groups` takes a `--path-prefix` rather than `--path`, so perhaps we could use `--skip-path-prefix`?

On the other hand, I'm wondering if we're being unnecessarily specific by adding a flag specifically for excluding prefixes — could we take a more general approach, something like `--exclude '/aws-reserved/*'`?

Comment (andrewjhumphrey) [reply iamy.go]: Very much prefer the `-prefix` addition to the argument name over `--exclude`, way too much ambiguity in that for my comfort.

Comment (andrewjhumphrey) [reply iamy.go]: Pushed a change to reflect this a3548ed

**Review by liamdawson — COMMENTED** [review]
I'm not really able to review the functionality, but the interface (and intended behaviour) LGTM

File: iamy/aws.go [hunk 2]
```diff
@@ -434,6 +435,14 @@ func (a *AwsFetcher) isSkippableManagedResource(cfnType CfnResourceType, resourc
 		}
 	}
 
+	if len(a.SkipPathPrefixes) > 0 {
```
Comment (patrobinson) [inline iamy/aws.go]: This if block is unnecessary, since the for loop below will just do nothing if the length is zero.

File: iamy/aws.go [hunk 3]
```diff
@@ -434,6 +435,14 @@ func (a *AwsFetcher) isSkippableManagedResource(cfnType CfnResourceType, resourc
...
+		for _, path := range a.SkipPathPrefixes {
```
Comment (patrobinson) [inline iamy/aws.go]: Can we use this to replace the code below that checks for `aws-service-role`? Given that's doing the same thing

**Review by patrobinson — DISMISSED** [review]

**Comment by patrobinson** [comment]
Should this flag also apply to the pull command?

**Comment by andrewjhumphrey** [comment]
> Should this flag also apply to the pull command?

Unless I've pooched something it should do the same thing for both push and pull..

File: iamy/aws.go (hunk 2 above)
Comment (andrewjhumphrey) [reply iamy/aws.go]: Too much C code in my background, fixed this one and the one just above it in 05872ed

File: iamy/aws.go (hunk 3 above)
Comment (andrewjhumphrey) [reply iamy/aws.go]: I didn't want to do that primary for clarity about why things are skipped.  Theoretically we could replace all of the `AWSServiceRole` skipping stuff with a more general purpose --skip-name-pattern or something like that.    I'll tackle that in the next PR.

**Review by patrobinson — APPROVED** [review]
